### ISSUE TITLE
Execution hangs on `sleep_us(1)` on Pico W

### ISSUE THREAD

**Issue by cshaa** [ArmDeveloperEcosystem/st7789-library-for-pico#5]
On my Raspberry Pi Pico W using the latest SDK, trying to use the library leads to crashing the CPU – execution just hangs, and the chip doesn't do anything else. Here's my project that reproduces the bug: [pico-st7789-bug](https://github.com/m93a/pico-st7789-bug). (This is happening regardless of whether there's any display connected to the Pico.)

I found out that it's caused by the `sleep_us(1)` calls at various places in the code (lines: [28](https://github.com/ArmDeveloperEcosystem/st7789-library-for-pico/blob/main/src/st7789.c#L28), [33](https://github.com/ArmDeveloperEcosystem/st7789-library-for-pico/blob/main/src/st7789.c#L33), [38](https://github.com/ArmDeveloperEcosystem/st7789-library-for-pico/blob/main/src/st7789.c#L38), [40](https://github.com/ArmDeveloperEcosystem/st7789-library-for-pico/blob/main/src/st7789.c#L40), [45](https://github.com/ArmDeveloperEcosystem/st7789-library-for-pico/blob/main/src/st7789.c#L45), [50](https://github.com/ArmDeveloperEcosystem/st7789-library-for-pico/blob/main/src/st7789.c#L50), [159](https://github.com/ArmDeveloperEcosystem/st7789-library-for-pico/blob/main/src/st7789.c#L159), [164](https://github.com/ArmDeveloperEcosystem/st7789-library-for-pico/blob/main/src/st7789.c#L164), [170](https://github.com/ArmDeveloperEcosystem/st7789-library-for-pico/blob/main/src/st7789.c#L170), [175](https://github.com/ArmDeveloperEcosystem/st7789-library-for-pico/blob/main/src/st7789.c#L175)). It seems that there is a race condition in the SDK that leads to missing the interrupt for the short timer and hanging indefinitely – potentially only on Pico W devices. Related issues: https://github.com/raspberrypi/pico-sdk/issues/1552, https://github.com/raspberrypi/pico-sdk/issues/1500, https://github.com/micropython/micropython/issues/12873

If I remove all these `sleep_us` calls, the library starts working as expected, and I'm able to run the blink example on a 240x240 display. If there isn't sufficient rationale for the `sleep_us(1)` calls, I advise to remove them altogether – at least for me the library works perfectly without them. However, if they are necessary for correct timing, we can replace them with something like [`busy_wait_at_least_cycles`](https://www.raspberrypi.com/documentation/pico-sdk/runtime.html#ga49c4fc3b5d3fbe9f2f6f7bc500b9db4f), and if they're a walkaround for [`spi_write_blocking` not blocking enough](https://forums.raspberrypi.com/viewtopic.php?t=364760), they can be replaced with a busywait until SPI is done:
```c
while (((spi_hw_t *)st7789_cfg.spi)->sr & SPI_SSPSR_BSY_BITS)
{
  tight_loop_contents();
}
```
I can make a PR with any of the proposed solutions.

**Comment by TheCustomFHD** [comment]
I will cross-test this with my screen aswell during the weekend, would be interesting if this works for me too.

**Comment by TheCustomFHD** [comment]
Soo.. it wasnt exactly uh.. "that weekend", sorry for that. BUT! i can confirm, that with a `sleep_us(10)`, my display functions. It also functions without the `sleep_us(1)`or `sleep_us(10)` waits. This is very nice for me, and huge thanks to @m93a for figuring this out. Now my old Rapsberry Pi 1 A Screens have a new nice usecase as a screen for my Pico's, amazing!

**Comment by cshaa** [comment]
@stale As far as I know, this is still an issue.

**Comment by TheCustomFHD** [comment]
I can confirm, this is still broken. Waiting for a response from what the Repository Owner wants to do

**Comment by cshaa** [comment]
Issues usually don't get fixed by ignoring them.

**Comment by TheCustomFHD** [comment]
Indeed, this fix is essentially free (it takes less than 3 minutes to implement). Although this library is painfully slow anyway. Iirc, the screen/driver supports to dump the whole framebuffer into it at once, which would be lots faster (but consume more RAM ofc)
